### PR TITLE
Use lxc's load config instead of our own crappy config parsing

### PR DESCRIPTION
**Pull request by tych0**
Signed-off-by: Tycho Andersen <tycho.andersen@canonical.com>

### CODE REVIEW

File: lxd/containers.go [hunk 1]
```diff
@@ -757,27 +757,20 @@ func (d *lxdContainer) applyConfig(config map[string]string) error {
...
-			}
-
-			key = strings.Trim(key, " ")
-			value = strings.Trim(value, " ")
-
-			// empty lines
-			if key == "" {
-				continue
-			}
+		f, err := ioutil.TempFile("", "tmpConfig")
```
Comment (stgraber) [inline lxd/containers.go]: Sorry for being picky on that, but my /tmp is usually a mess after a while so I like things prefixed with something I can understand :)

Can you change this to lxd_config or something like that?

Bonus point if you can change the one in images.go too (Serge already took care of those in the client code).

Comment (tych0) [inline lxd/containers.go]: On Wed, Mar 11, 2015 at 09:28:25AM -0700, St�phane Graber wrote:
> > -		for _, line := range strings.Split(lxcConfig, "\n") {
> > -			contents := strings.SplitN(line, "=", 2)
> > -			key := contents[0]
> > -			value := ""
> > -			if len(contents) > 1 {
> > -				value = contents[1]
> > -			}
> > -
> > -			key = strings.Trim(key, " ")
> > -			value = strings.Trim(value, " ")
> > -
> > -			// empty lines
> > -			if key == "" {
> > -				continue
> > -			}
> > +		f, err := ioutil.TempFile("", "tmpConfig")
> 
> Sorry for being picky on that, but my /tmp is usually a mess after a while so I like things prefixed with something I can understand :)
> 
> Can you change this to lxd_config or something like that?
> 
> Bonus point if you can change the one in images.go too (Serge already took care of those in the client code).

No worries, that seems more polite anyway. I've pushed the fix.

File: lxd/containers.go [hunk 2]
```diff
@@ -757,27 +757,20 @@ func (d *lxdContainer) applyConfig(config map[string]string) error {
...
-			}
-
-			key = strings.Trim(key, " ")
-			value = strings.Trim(value, " ")
-
-			// empty lines
-			if key == "" {
-				continue
-			}
+		f, err := ioutil.TempFile("", "lxd_config")
```
Comment (stgraber) [inline lxd/containers.go]: Does TempFile add a separate (_ or something else) for us there?

A grep through our current code shows that we've been using a trailing underscore in our prefixes so far.

Comment (tych0) [inline lxd/containers.go]: On Wed, Mar 11, 2015 at 10:16:45AM -0700, St�phane Graber wrote:
> > -		for _, line := range strings.Split(lxcConfig, "\n") {
> > -			contents := strings.SplitN(line, "=", 2)
> > -			key := contents[0]
> > -			value := ""
> > -			if len(contents) > 1 {
> > -				value = contents[1]
> > -			}
> > -
> > -			key = strings.Trim(key, " ")
> > -			value = strings.Trim(value, " ")
> > -
> > -			// empty lines
> > -			if key == "" {
> > -				continue
> > -			}
> > +		f, err := ioutil.TempFile("", "lxd_config")
> 
> Does TempFile add a separate (_ or something else) for us there?
> 
> A grep through our current code shows that we've been using a trailing underscore in our prefixes so far.

It doesn't, it just adds a number at the end. We're using trailing
underscores in some places and not others.

Comment (stgraber) [inline lxd/containers.go]: Looks to me like we've been using the trailing underscore everywhere until now:

    stgraber@dakara:~/data/code/lxc/lxd$ grep -r TempFile
    lxc/config.go:	f, err := ioutil.TempFile("", "lxc_profile_")
    lxc/image.go:		f, err := ioutil.TempFile("", "lxc_image_")
    lxd/images.go:	f, err := ioutil.TempFile(dirname, "image_")

Comment (tych0) [inline lxd/containers.go]: On Wed, Mar 11, 2015 at 10:32:33AM -0700, St�phane Graber wrote:
> > -		for _, line := range strings.Split(lxcConfig, "\n") {
> > -			contents := strings.SplitN(line, "=", 2)
> > -			key := contents[0]
> > -			value := ""
> > -			if len(contents) > 1 {
> > -				value = contents[1]
> > -			}
> > -
> > -			key = strings.Trim(key, " ")
> > -			value = strings.Trim(value, " ")
> > -
> > -			// empty lines
> > -			if key == "" {
> > -				continue
> > -			}
> > +		f, err := ioutil.TempFile("", "lxd_config")
> 
> Looks to me like we've been using the trailing underscore everywhere until now:
> 
>     stgraber@dakara:~/data/code/lxc/lxd$ grep -r TempFile
>     lxc/config.go:	f, err := ioutil.TempFile("", "lxc_profile_")
>     lxc/image.go:		f, err := ioutil.TempFile("", "lxc_image_")
>     lxd/images.go:	f, err := ioutil.TempFile(dirname, "image_")

criu:~/packages/go/src/github.com/lxc/lxd master grep -ri tempdir *
lxd/migration/migrate.go: checkpointDir, err := ioutil.TempDir("", "lxd_migration")
lxd/migration/migrate.go: imagesDir, err := ioutil.TempDir("", "lxd_migration")
lxd/migration/rsync_test.go:  source, err := ioutil.TempDir("", "source")
lxd/migration/rsync_test.go:  sink, err := ioutil.TempDir("", "sink")

Not for TempDir, though.

Comment (stgraber) [inline lxd/containers.go]: indeed... I wish things were a bit more consistent...

Personally I'm in favor of the trailing underscore as a separator and of using the same kind of naming everywhere, so the TempDir test ones should probably become lxd_test_source_ and lxd_test_sink_ or something like that.

Anyway, bikeshedding :)

**Pull request merged**